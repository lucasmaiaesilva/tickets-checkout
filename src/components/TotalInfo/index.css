.total-info {
  margin-top: 14px;
}

.total-info .title-text {
  margin: 0;
}

.total-info .extra-info {
  color: #777;
  font-size: 0.8rem;
}

@media screen and (min-width: 734px) {
  .total-info {
    display: flex;
    justify-content: space-between;
  }
}
