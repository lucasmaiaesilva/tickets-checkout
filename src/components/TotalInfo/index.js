import React from 'react';
import TitleText from '../TitleText';

import './index.css';

const TotalInfo = ({ title, price, extra, withDiscount }) => (
  <div className={`total-info ${withDiscount ? 'discount' : false}`}>
    <TitleText>{ title }</TitleText>
    <div>
      <TitleText>{ price }</TitleText>
      {extra && <div className="extra-info">{ extra }</div>}
    </div>
  </div>
);

export default TotalInfo;

