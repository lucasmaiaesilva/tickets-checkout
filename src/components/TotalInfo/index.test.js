
import React from 'react';
import { shallow } from 'enzyme';
import TotalInfo from './';

describe('[Component] TotalInfo', () => {
  const commomProps = {
    title: 'Example of title',
    price: '1.000,00'
  };
  it('render correctly with whole props', () => {
    const wrapper = shallow(<TotalInfo {...commomProps} extra="extra text" />);
    expect(wrapper).toMatchSnapshot();
  })
  it('render correctly without extra text', () => {
    const wrapper = shallow(<TotalInfo {...commomProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
