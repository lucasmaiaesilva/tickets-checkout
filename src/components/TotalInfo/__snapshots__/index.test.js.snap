// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] TotalInfo render correctly with discount flag 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalInfo
    price="1.000,00"
    title="Example of title"
    withDiscount={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TitleText>
          Example of title
</TitleText>,
        <div>
          <TitleText>
                    1.000,00
          </TitleText>
</div>,
      ],
      "className": "total-info discount",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Example of title",
        },
        "ref": null,
        "rendered": "Example of title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TitleText>
              1.000,00
</TitleText>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1.000,00",
            },
            "ref": null,
            "rendered": "1.000,00",
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TitleText>
            Example of title
</TitleText>,
          <div>
            <TitleText>
                        1.000,00
            </TitleText>
</div>,
        ],
        "className": "total-info discount",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Example of title",
          },
          "ref": null,
          "rendered": "Example of title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TitleText>
                1.000,00
</TitleText>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1.000,00",
              },
              "ref": null,
              "rendered": "1.000,00",
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`[Component] TotalInfo render correctly with whole props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalInfo
    extra="extra text"
    price="1.000,00"
    title="Example of title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TitleText>
          Example of title
</TitleText>,
        <div>
          <TitleText>
                    1.000,00
          </TitleText>
          <div
                    className="extra-info"
          >
                    extra text
          </div>
</div>,
      ],
      "className": "total-info false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Example of title",
        },
        "ref": null,
        "rendered": "Example of title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TitleText>
              1.000,00
</TitleText>,
            <div
              className="extra-info"
>
              extra text
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1.000,00",
            },
            "ref": null,
            "rendered": "1.000,00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "extra text",
              "className": "extra-info",
            },
            "ref": null,
            "rendered": "extra text",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TitleText>
            Example of title
</TitleText>,
          <div>
            <TitleText>
                        1.000,00
            </TitleText>
            <div
                        className="extra-info"
            >
                        extra text
            </div>
</div>,
        ],
        "className": "total-info false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Example of title",
          },
          "ref": null,
          "rendered": "Example of title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TitleText>
                1.000,00
</TitleText>,
              <div
                className="extra-info"
>
                extra text
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1.000,00",
              },
              "ref": null,
              "rendered": "1.000,00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "extra text",
                "className": "extra-info",
              },
              "ref": null,
              "rendered": "extra text",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`[Component] TotalInfo render correctly without extra text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalInfo
    price="1.000,00"
    title="Example of title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TitleText>
          Example of title
</TitleText>,
        <div>
          <TitleText>
                    1.000,00
          </TitleText>
</div>,
      ],
      "className": "total-info false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Example of title",
        },
        "ref": null,
        "rendered": "Example of title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TitleText>
              1.000,00
</TitleText>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1.000,00",
            },
            "ref": null,
            "rendered": "1.000,00",
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TitleText>
            Example of title
</TitleText>,
          <div>
            <TitleText>
                        1.000,00
            </TitleText>
</div>,
        ],
        "className": "total-info false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Example of title",
          },
          "ref": null,
          "rendered": "Example of title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TitleText>
                1.000,00
</TitleText>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1.000,00",
              },
              "ref": null,
              "rendered": "1.000,00",
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
